### ISSUE TITLE
CA2000 reported on every using statement with IAsyncDisposable

### ISSUE THREAD

**Issue by cytoph** [dotnet/roslyn-analyzers#6512]
### Analyzer

**Diagnostic ID**: [CA2000](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2000): `Dispose objects before losing scope`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 7.0.0 (Latest Stable)

(The bug also exists in all versions of the analyzer built-in into the SDK in 5.0, 6.0 and 7.0)

### Describe the bug

Whenever one uses a using statement with an `IAsyncDisposable` the analyzer falsely reports CA2000.

### Steps To Reproduce

OPTIONAL: If you want, create a custom implementation of `IAsyncDisposable`. For this example I'll use `FileStream` to show that the bug affects even the most common built-in types.

1. Create an async method that contains a async using statement (like shown belown)
2. See error

```
public async Task DoSomethingAsync()
{
    await using var stream = new FileStream("C:\somefile.txt", FileMode.OpenOrCreate).ConfigureAwait(false); // this already shows the false issue
}
```

### Expected behavior

`await using var stream = new FileStream("C:\somefile.txt", FileMode.OpenOrCreate).ConfigureAwait(false);` will not report CA2000. Maybe if not used with a `using` it should report an issue, but that should be a newly created one, not CA2000, because CA2000 explicitly states one should call `System.IDisposable.Dispose` which is not correct for `IAsyncDisposable`.

### Actual behavior

`await using var stream = new FileStream("C:\somefile.txt", FileMode.OpenOrCreate).ConfigureAwait(false);` reports CA2000. `IAsyncDisposable` in general reports this, no matter if used in a `using` or not. The only way to suppress the issue is to initialize a variable and after that start a try-finally-block where the variable is disposed in the `finally` part, which is exactly what the `using` is for in C#.

## Additional context

This bug was already reported in issue #3042 but no one bothered to re-open it even though it was **not** fixed - so what other way to call attention to this bug than creating a new issue? 🤷‍♂️

**Comment by alexrp** [comment]
I'm also regularly running into this in a project using .NET 8 Preview 3.
